### ISSUE TITLE
Unexpected behavior resolving collections, if ansible-lint is executed outside project dir

### ISSUE THREAD

**Issue by xrow** [ansible/ansible-lint#3957]
##### Summary

Issue resolving paths to local collections when ansible-lint is not executed in project dir. It loos like tit will not alter or change the path from collections_path with is ./.

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
[root@localhost ansible]# ansible-lint --version
ansible-lint 6.22.0 using ansible-core:2.15.6 ansible-compat:4.1.10 ruamel-yaml:None ruamel-yaml-clib:None
A new release of ansible-lint is available: 6.22.0 → 6.22.1 Upgrade by running: pip install --upgrade ansible-lint
```
- ansible installation method: pip
- ansible-lint installation method: pip

##### STEPS TO REPRODUCE

have a local collection with collections_path = ~/.ansible/collections:/usr/share/ansible/collections:./

```console (paste below)
[root@localhost infrastructure]# ansible-lint ansible/
WARNING  Listing 4 violation(s) that are fatal
syntax-check[specific]: the role 'xxx.infrastructure.vault' was not found in /scm/infrastructure/ansible/ansible_collections/xxx/infrastructure/playbooks/roles:/root/.cache/ansible-compat/9fec08/roles:/root/.ansible/roles:/usr/share/ansible/roles:/etc/ansible/roles:/scm/infrastructure/ansible/ansible_collections/xxx/infrastructure/playbooks
ansible/ansible_collections/xxx/infrastructure/playbooks/local.yml:18:15

syntax-check[missing-file]: Unable to retrieve file contents
ansible/playbooks/add-chart.yml:1:1 Could not find or access '/scm/infrastructure/ansible/playbooks/xxx.infrastructure.local' on the Ansible Controller.

syntax-check[missing-file]: Unable to retrieve file contents
ansible/playbooks/build.yml:1:1 Could not find or access '/scm/infrastructure/ansible/playbooks/xxx.infrastructure.local' on the Ansible Controller.

syntax-check[missing-file]: Unable to retrieve file contents
ansible/playbooks/clean-all.yml:1:1 Could not find or access '/scm/infrastructure/ansible/playbooks/xxx.infrastructure.local' on the Ansible Controller.


                    Rule Violation Summary                     
 count tag                        profile rule associated tags 
     3 syntax-check[missing-file] min     core, unskippable    
     1 syntax-check[specific]     min     core, unskippable    

Failed: 4 failure(s), 0 warning(s) on 43 files.
A new release of ansible-lint is available: 6.22.0 → 6.22.1 Upgrade by running: pip install --upgrade ansible-lint
[root@localhost infrastructure]# ansible-lint -p ansible/
WARNING  Listing 4 violation(s) that are fatal
ansible/ansible_collections/xxx/infrastructure/playbooks/local.yml:18:15: syntax-check[specific]: the role 'xxx.infrastructure.vault' was not found in /scm/infrastructure/ansible/ansible_collections/xxx/infrastructure/playbooks/roles:/root/.cache/ansible-compat/9fec08/roles:/root/.ansible/roles:/usr/share/ansible/roles:/etc/ansible/roles:/scm/infrastructure/ansible/ansible_collections/xxx/infrastructure/playbooks
ansible/playbooks/add-chart.yml:1:1: syntax-check[missing-file]: Unable to retrieve file contents
ansible/playbooks/build.yml:1:1: syntax-check[missing-file]: Unable to retrieve file contents
ansible/playbooks/clean-all.yml:1:1: syntax-check[missing-file]: Unable to retrieve file contents

                    Rule Violation Summary                     
 count tag                        profile rule associated tags 
     3 syntax-check[missing-file] min     core, unskippable    
     1 syntax-check[specific]     min     core, unskippable    

Failed: 4 failure(s), 0 warning(s) on 43 files.
A new release of ansible-lint is available: 6.22.0 → 6.22.1 Upgrade by running: pip install --upgrade ansible-lint
[root@localhost infrastructure]# cd ansible/ && ansible-lint .
WARNING  Skipped installing collection dependencies due to running in offline mode.

Passed: 0 failure(s), 0 warning(s) on 5 files. Last profile that met the validation criteria was 'production'.
A new release of ansible-lint is available: 6.22.0 → 6.22.1 Upgrade by running: pip install --upgrade ansible-lint
```

##### Desired Behavior

ansible-lint -p ansible/ shall work the same as cd ansible/ && ansible-lint .

Possible security bugs should be reported via email to `security@ansible.com`

##### Actual Behavior

See the /scm/infrastructure/ansible missing in the failed run.

```paste below
[root@localhost ansible]# ansible-lint -p . -vvvv
INFO     Set ANSIBLE_COLLECTIONS_PATH=/root/.cache/ansible-compat/0cb87f/collections:/root/.ansible/collections:/usr/share/ansible/collections:/scm/infrastructure/ansible
[root@localhost infrastructure]# ansible-lint -p ansible/ -vvvv
INFO     Set ANSIBLE_COLLECTIONS_PATH=/root/.cache/ansible-compat/9fec08/collections:/root/.ansible/collections:/usr/share/ansible/collections
```

[minimum complete verifiable example]: http://stackoverflow.com/help/mcve

**Comment by ssbarnea** [comment]
I think that this is not a bug because when you pass `-p` you force ansible lint to assume a different project instead of the detected one. Different is to be expected.

**Comment by zac90** [comment]
I too am experiencing a similar issue.

I execute `ansible-lint` from the root dir of the project, it then traverses into a number of different roles which each have molecule files in them. It then fails on each `molecule/prepare.yml` saying `syntax-check[specific]: the role 'xyz.common.rolename' was not found in <big list of search locations>` . Yet this collection has already been fetched as apart of the intilization. To get it to run I have to use `--exclude molecule/*/prepare.yml` otherwise it keeps failing on the syntax-check stage.

**Comment by garymm** [comment]
I'm also facing this. I also tried setting `--config-file`. E.g.:
```bash
ansible-lint --config-file=./ansible/.config/ansible-lint.yml
```

But ansible-lint insists on going up to the location of the `.git` folder rather than the parent directory of the config file (as seen from the log message `Identified <repo root> as project root due .git directory.`

Has anyone found a work-around?
It's annoying because pre-commit doesn't easily allow changing the CWD when running `ansible-lint`.

**Comment by cardoe** [comment]
@garymm have you found a workaround for this?

**Comment by bdsoha** [comment]
@garymm Any updates?

**Comment by ssbarnea** [comment]
> I'm also facing this. I also tried setting `--config-file`. E.g.:
> 
> ansible-lint --config-file=./ansible/.config/ansible-lint.yml
> But ansible-lint insists on going up to the location of the `.git` folder rather than the parent directory of the config file (as seen from the log message `Identified <repo root> as project root due .git directory.`
> 
> Has anyone found a work-around? It's annoying because pre-commit doesn't easily allow changing the CWD when running `ansible-lint`.

Have you tried `--project-dir` option?

**Comment by cardoe** [comment]
> > I'm also facing this. I also tried setting `--config-file`. E.g.:
> > 
> > ansible-lint --config-file=./ansible/.config/ansible-lint.yml
> > But ansible-lint insists on going up to the location of the `.git` folder rather than the parent directory of the config file (as seen from the log message `Identified <repo root> as project root due .git directory.`
> > 
> > Has anyone found a work-around? It's annoying because pre-commit doesn't easily allow changing the CWD when running `ansible-lint`.
> 
> Have you tried `--project-dir` option?

Doesn't work for me.

**Comment by dbrennand** [comment]
Hi all, I think I've just hit this behaviour and IMHO, this behaviour feels like a bug. I was expecting that when `--project-dir` is specified the collection itself is installed. This is what happens when you run `ansible-lint` from inside the collection directory, but if you're in the parent directory and use `--project-dir`, it is not installed and you get errors about roles not being found.

I created a reproducible [example](https://github.com/dbrennand/repro/tree/collection) repository.

```
repro/ansible on  collection via 🐍 v3.13.1 (repro)
❯ ansible-lint --version
ansible-lint 25.6.0 using ansible-core:2.18.6 ansible-compat:25.5.0 ruamel-yaml:0.18.14 ruamel-yaml-clib:0.2.12
```

## Collection Structure

The collection is in the `ansible` subdirectory.

```
repro on  collection is 📦 v0.1.0 via 🐍 v3.13.1 (repro)
❯ tree -L 1 ansible
ansible
├── CHANGELOG.rst
├── changelogs
├── CODE_OF_CONDUCT.md
├── CONTRIBUTING
├── devfile.yaml
├── docs
├── extensions
├── galaxy.yml
├── LICENSE
├── MAINTAINERS
├── meta
├── playbooks
├── plugins
├── pyproject.toml
├── README.md
├── requirements.txt
├── roles
├── test-requirements.txt
├── tests
└── tox-ansible.ini

9 directories, 12 files
```

## Inside the collection directory itself

```
repro/ansible on  collection via 🐍 v3.13.1 (repro)
❯ ansible-lint -v
INFO     Identified /Users/dab/repos/repro as project root due .git directory.
INFO     Collection paths was patched to include extra directories /Users/dab/.ansible/collections,/usr/share/ansible/collections,/Users/dab/repos/repro/.venv/lib/python3.13/site-packages
INFO     Set ANSIBLE_LIBRARY=/Users/dab/repos/repro/ansible/.ansible/modules:plugins/modules:/Users/dab/.ansible/plugins/modules:/usr/share/ansible/plugins/modules
INFO     Set ANSIBLE_COLLECTIONS_PATH=/Users/dab/repos/repro/ansible/.ansible/collections:/Users/dab/.ansible/collections:/usr/share/ansible/collections:/Users/dab/repos/repro/.venv/lib/python3.13/site-packages
INFO     Set ANSIBLE_ROLES_PATH=/Users/dab/repos/repro/ansible/.ansible/roles:roles:/Users/dab/.ansible/roles:/usr/share/ansible/roles:/etc/ansible/roles
INFO     Provisioning collection ansible.utils:* from galaxy.yml
INFO     Running from /Users/dab/repos/repro/ansible : ansible-galaxy collection install -vvv ansible.utils:*
INFO     Provisioning collection ansible.utils:* from galaxy.yml
INFO     Running from /Users/dab/repos/repro/ansible : ansible-galaxy collection install -vvv ansible.utils:*
------> Here: INFO     Running from /Users/dab/repos/repro/ansible : ansible-galaxy collection install -vvv --force /Users/dab/repos/repro/ansible
INFO     Loading ignores from .gitignore
INFO     Loading ignores from tests/.gitignore
INFO     Loading ignores from .gitignore
INFO     Loading ignores from tests/.gitignore
INFO     Executing syntax check on playbook playbooks/example.yml (0.24s)
INFO     Executing syntax check on playbook extensions/molecule/utils/playbooks/noop.yml (0.24s)
INFO     Executing syntax check on playbook extensions/molecule/utils/playbooks/converge.yml (0.24s)
INFO     Executing syntax check on role roles/run (0.24s)
WARNING  The following filters were mocked during the run: dbrennand.repro.sample_filter

Passed: 0 failure(s), 0 warning(s) on 77 files. Last profile that met the validation criteria was 'production'.
```

## Using `--project-dir`

```
repro on  collection is 📦 v0.1.0 via 🐍 v3.13.1 (repro)
❯ uv run ansible-lint -v --project-dir "$(pwd)/ansible"
INFO     Collection paths was patched to include extra directories /Users/dab/.ansible/collections,/usr/share/ansible/collections,/Users/dab/repos/repro/.venv/lib/python3.13/site-packages
INFO     Set ANSIBLE_LIBRARY=/Users/dab/repos/repro/.ansible/modules:/Users/dab/.ansible/plugins/modules:/usr/share/ansible/plugins/modules
INFO     Set ANSIBLE_COLLECTIONS_PATH=/Users/dab/repos/repro/.ansible/collections:/Users/dab/.ansible/collections:/usr/share/ansible/collections:/Users/dab/repos/repro/.venv/lib/python3.13/site-packages
INFO     Set ANSIBLE_ROLES_PATH=/Users/dab/repos/repro/.ansible/roles:/Users/dab/.ansible/roles:/usr/share/ansible/roles:/etc/ansible/roles
INFO     Provisioning collection ansible.utils:* from galaxy.yml
INFO     Running from /Users/dab/repos/repro : ansible-galaxy collection install -vvv ansible.utils:*
INFO     Loading ignores from .gitignore
INFO     Loading ignores from ansible/.gitignore
INFO     Loading ignores from ansible/tests/.gitignore
INFO     Loading ignores from .gitignore
INFO     Loading ignores from ansible/.gitignore
INFO     Loading ignores from ansible/tests/.gitignore
INFO     Executing syntax check on playbook ansible/playbooks/example.yml (0.22s)
INFO     Executing syntax check on playbook ansible/extensions/molecule/utils/playbooks/converge.yml (0.24s)
INFO     Executing syntax check on playbook ansible/extensions/molecule/utils/playbooks/noop.yml (0.24s)
INFO     Executing syntax check on role ansible/roles/run (0.24s)
WARNING  Listing 1 violation(s) that are fatal
syntax-check[specific]: the role 'dbrennand.repro.run' was not found in /Users/dab/repos/repro/ansible/playbooks/roles:/Users/dab/repos/repro/.ansible/roles:/Users/dab/.ansible/roles:/usr/share/ansible/roles:/etc/ansible/roles:/Users/dab/repos/repro/ansible/playbooks
ansible/playbooks/example.yml:7:7

WARNING  The following filters were mocked during the run: dbrennand.repro.sample_filter

# Rule Violation Summary

  1 syntax-check profile:min tags:core,unskippable

Failed: 1 failure(s), 0 warning(s) on 75 files.
```
